syntax="proto2";
import "pds_option.proto";

package baidu.infinite.map_rnn;
//package baidu.paddle_serving.predictor.map_rnn;

option cc_generic_services = true;

/*message Tensor {
    required string name = 1;
    repeated uint32 shape = 2;
    required bytes features = 3;
};*/

message Line {
	required bytes value = 1;
};

message Lines {
	repeated Line line = 1;
};

message DenseInstance {
    repeated Lines step_data = 1;
	required Lines week_data = 2;
	required Lines minute_data = 3;
	required uint32 batch_size = 4;
};

message Request {
    repeated DenseInstance instances = 1;
};

message DensePrediction {
    repeated float categories = 1;
};

message Response {
    repeated DensePrediction predictions = 1;
};

service MapRnnService {
    rpc inference(Request) returns (Response);
    rpc debug(Request) returns (Response);
    //option (pds.options).generate_impl = true;
    option (pds.options).generate_stub = true;
};
